### DISCUSSION TITLE
PID's for EV's

### DISCUSSION THREAD

**Discussion by jay-oswald** [meatpiHQ/wican-fw#166] [General]
https://github.com/Krolow93/OBD-PIDs-for-HKMC-EVs/tree/master

I found this repository via the ABRP app, has many, many PID's and equations for various EV's

**Comment by Gics78** [comment]
To make my day a litte productive: This are some PIDs for a Genesis eGV70 EV. It should be the same for Ioniq 5 or equal. Have fun!#
![grafik](https://github.com/user-attachments/assets/67f4da52-a9d9-41fc-bdc5-65cbf989e861)

**Reply by tung256** [reply]
i can confirm some of the equations work for my Ioniq5 too.  here's the equation variable conversion to "B" that i mapped out. hope it helps others as well. 
![image](https://github.com/user-attachments/assets/3c48f446-d0fa-4f0c-a33d-430391cce5d3)

here's one equation for State of Health that i dont understand:  ((z<8)+aa)/10   using my map, i get ((B34<8)+B35)/10
do you know what  this operation does   "<"   ???  what does B34 do to 8?

**Reply by njh** [reply]
The `<8` operator might rotate 8-bits to the left. In some languages it is `<<`.

**Reply by tung256** [reply]
very cool! i had no idea that's a thing.
regardless, i tried that equation for my car and it does not work :(
i found this google sheets with all the equations but some things like tire pressure,door open or closed, door locked or unlocked  is not there https://docs.google.com/spreadsheets/d/1J4LqumlVlKGymRWc2U0VRJjQFK7NDF5TPUfb1ph5Aow/edit?usp=sharing

**Reply by jay-oswald** [reply]
@tung256 if its working for you with SOH, feel free to do a PR, to add it to the new vehicle profiles for the ioniq if its working locally for you

**Comment by njh** [comment]
This might be a good resource too:
* https://docs.openvehicles.com/en/latest/introduction.html#vehicle-support
* https://github.com/openvehicles/Open-Vehicle-Monitoring-System-3/tree/master/vehicle/OVMS.V3/components
